junction/proposals/management/commands/export_conf_proposals.py

```diff
@@ -0,0 +1,80 @@
+
+        parser.add_argument('--user_id', default=None, help='Enter your user id')
+
+    def handle(self, *args, **options):
+
+        try:
+            conference = get_object_or_404(Conference, slug=self.options.get('conference_slug'))
+            user = User.objects.get(id=self.options.get('user_id'))
+
+            if not is_conference_moderator(user=user, conference=conference):
+                raise PermissionDenied
+
+            proposal_sections = conference.proposal_sections.all()
+            proposals_qs = Proposal.objects.select_related('proposal_type', 'proposal_section', 'conference', 'author').filter(conference=conference, status=ProposalStatus.PUBLIC)
+            proposals_qs = sorted(proposals_qs, key=lambda x: x.get_reviewer_votes_sum(), reverse=True)
+            vote_values_list = ProposalSectionReviewerVoteValue.objects.order_by('-vote_value')
+            vote_values_list = [v.vote_value for v in vote_values_list]
+            vote_values_desc = tuple(i.description for i in ProposalSectionReviewerVoteValue.objects.order_by('-vote_value'))
+            header = ('Proposal Type', 'Title', 'Sum of reviewer votes', 'No. of reviewer votes') + ('Public votes count', 'Vote comments')+ tuple(vote_values_desc)
+
+            csv_contents = []
+
+            for section in proposal_sections:
+                section_proposals = [p for p in proposals_qs if p.proposal_section == section]
+
+                for index, p in enumerate(section_proposals, 1):
+                    vote_details = tuple(p.get_reviewer_votes_count_by_value(v) for v in vote_values_list)
+                    vote_comment = '\n'.join([comment.comment for comment in p.proposalcomment_set.filter(vote=True, deleted=False)])
+                    row = {header[0]: p.proposal_type.name,
+                           header[1]: p.title,
+                           header[2]: p.get_reviewer_votes_sum(),
+                           header[3]: p.get_reviewer_votes_count(),
+                           header[4]: p.get_voutes_count(),
+                           header[5]: vote_comment}
+                    for i in range(len(tuple(vote_values_desc))):
+                        row[tuple(vote_values_desc)[i]] = vote_details[i]
+
+                    csv_contents.append(row)
+
+            csv_file_name = "%s-%s" % (user.name, conference.name)
+            csv_file_location = join(ROOT_DIR, 'csvs', csv_file_name)
+
+            with open(csv_file_location, 'w') as csvfile:
+                writer = csv.DictWriter(csvfile, fieldnames=header)
+
+                writer.writeheader()
+                for row in csv_contents:
+                    writer.writerow(row)
+
+            self.stdout.write("Successfully created the csv file")
+        except User.DoesNotExist:
+            self.stdout.write("Invalid user")
+        except Conference.DoesNotExist:
```

Comment:
I think you could catch this exceptions where you're making queries on the objects and stdout the message and return from there. No need to go through all the code if they failed.

Reply:
The exceptions should move to the end, right? If the exception is caught earlier, it won't go ahead with the `try...` code, and directly skip to the exception.

Reply:
What will be the point of going through all the code if you already have an exception? E.g. In this case what will be the point of going through all the code if you don't have `conference`?

Note that you're catching specific exceptions that means you already know in where they might be getting raised so I think it would be good idea to catch them there and return.

Reply:
Handled